src/polyfem/main.cpp

```diff
@@ -215,7 +215,7 @@ int optimization_simulation(const CLI::App &command_line,
 	OptState opt_state;
 	opt_state.init(opt_args, is_strict);
 
-	opt_state.create_states(opt_state.args["compute_objective"].get<bool>() ? polyfem::solver::CacheLevel::Solution : polyfem::solver::CacheLevel::Derivatives);
+	opt_state.create_states(opt_state.args["compute_objective"].get<bool>() ? polyfem::solver::CacheLevel::Solution : polyfem::solver::CacheLevel::Derivatives, max_threads);
```

Comment:
@Huangzizhou you changed this before, is it ok now?

Reply:
This is another issue that I didn't notice, should be fixed

Reply:
So can I merge like this or do you want to fix something @Huangzizhou?

Reply:
oh wait this fix has a small issue, let me fix it